### PR TITLE
Use pre-train output scaler to scale training data in CLI

### PR DESCRIPTION
**Pull request by KnathanM**
## Description
Usually the target distribution of a training set is normalized to make learning it easier. The same normalization is applied to the validation set in Chemprop. This normalization is then saved as part of the model so that predictions can be transformed back to the unscaled space. 

In transfer learning, there are two training sets. The model's output unscale transform is not updated by the fine-tuning dataset, so we need to scale the fine-tuning targets to match the model's unscale transform. 

## Current workflow
Currently we normalize the fine-tuning training set based on its own targets and then use that normalization for the fine-tuning validation set. This normalization is then thrown away when we load in a pre-trained model. So the targets in fine-tune sets are normalized differently than what the model knows, so the model can't unnormalize them.

## Desired workflow
It is a bit difficult to get the model's unscale transform, which we need for scaling the fine-tune sets. We build the data loaders before and separate from building the model, so we have to load the model twice. There is also the case that the original regression model didn't scale its training data and an identity function is used instead. 

## Questions
I use the output_scaler that is in the first model if multiple models are specified. We assume that all models are from the same ensemble:
https://github.com/chemprop/chemprop/blob/acd5fcb34d3038edb53ce7e2429e8f580b7cfb95/chemprop/cli/train.py#L970
Because of this, all the models should have the same output scaler. 

If there are multiple models because they were trained on different data splits, the CLI is not equipped to handle that. I'm in favor of saying we won't support that and just telling people they should run Chemprop manually multiple times if they want to do transfer learning with pre-training on different data splits.

### CODE REVIEW

**Review by shihchengli — COMMENTED** [review]
Thanks @KnathanM for making this PR. I also agree that we don’t support loading scalers for each model. We could write a warning message when multiple model paths are provided. I have two questions regarding this PR:
1. I think the `input_transforms` from the pre-trained model should also be loaded and used for the fine-tuning dataset.
2. What do you think about writing a `load_scalers` function to load both the `input_scaler` and `output_scaler`?

File: chemprop/cli/train.py [hunk 1]
```diff
@@ -1196,18 +1245,28 @@ def main(args):
...
-            logger.info(f"Train data: mean = {output_scaler.mean_} | std = {output_scaler.scale_}")
-            output_transform = UnscaleTransform.from_standard_scaler(output_scaler)
-        else:
+        if args.checkpoint or args.model_frzn is not None:
+            model_paths = find_models(args.checkpoint)
+            # assume all models have the same output scaling, meaning they were trained on the same data
+            model_path = model_paths[0] if args.checkpoint else args.model_frzn
+            load_and_use_pretrained_model_scalers(model_path, train_dset, val_dset)
+            input_transforms = (None, None, None)
             output_transform = None
```
Comment (shihchengli) [inline chemprop/cli/train.py:1257]: Just a note: they are set to `None` because the `input_transforms` and `output_transform` are only needed when building a new model from scratch.

**Review by shihchengli — APPROVED** [review]

File: chemprop/cli/train.py [hunk 2]
```diff
@@ -1197,37 +1245,28 @@ def main(args):
...
             save_smiles_splits(args, output_dir, train_dset, val_dset, test_dset)
 
-        if "regression" in args.task_type:
-            if args.checkpoint or args.model_frzn is not None:
-                model_paths = find_models(args.checkpoint)
-                mpnn_cls = (
-                    MulticomponentMPNN if isinstance(train_dset, MulticomponentDataset) else MPNN
+        if args.checkpoint or args.model_frzn is not None:
+            model_paths = find_models(args.checkpoint)
+            # assume all models have the same output scaling, meaning they were trained on the same data
```
[the remark this answers is not in the document]
Comment (KnathanM) [reply chemprop/cli/train.py]: Good idea, I tweaked the warning message a bit.
